### PR TITLE
Move event handling to AnimationController

### PR DESCRIPTION
**Pull request by kiliangrashoff**
Use the Guava EventBus to subscribe to (observe) events thrown from the model and handle their corresponding animations in the AnimationController.

This considerably reduces the complexity of both the GameController and the GUIController, fixing our incode warning. An explanation for using the EventBus instead of an ordinary observer pattern will be given in our report.

< feature

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java [hunk 1]
```diff
@@ -1,309 +1 @@
...

    /**
     * Update the displayed level.
     *
     * @param level the new level
     */
    public void updateLevel(int level) {
        levelView.update(level);
    }
}
```
Comment (kiliangrashoff) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java]: This diff looks weird, but as you can see when viewing the whole file the GUIController has not been deleted, and does have a newline at the end of the file.

File: src/main/java/nl/tudelft/ti2206/semgr8/model/data/EventBusSingleton.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Get the EventBus instance.
+     *
+     * @return the EventBus
+     */
+    public static EventBus getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+}
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/data/EventBusSingleton.java]: missing newline

File: src/main/java/nl/tudelft/ti2206/semgr8/model/PieceSpawner.java [hunk 3]
```diff
@@ -27,7 +30,7 @@ protected PieceSpawner(Board board) {
      * @param level the level the current game is in
      * @return the list of new gems that have spawned
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/PieceSpawner.java]: incorrect javadoc, also occurs in more places in this file

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java [hunk 4]
```diff
@@ -126,8 +114,8 @@ public SpawnData spawnGems(ArrayList<Gem> gems) {
      * @param level the level the current game is in
      * @return the list of new gems that have spawned
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java]: please update the javadoc in this file too.

File: src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java [hunk 5]
```diff
@@ -76,18 +85,16 @@ public void testEquals() {
     @Test
     public void testRemoveAllConnectedLines() {
```
Comment (albert-smit) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java]: test without assert

**Comment by albert-smit** [comment]
rest looks fine

File: src/main/java/nl/tudelft/ti2206/semgr8/model/data/EventBusSingleton.java (hunk 2 above)
Comment (kiliangrashoff) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/data/EventBusSingleton.java]: added

**Pull request merged**